src/components/LibraryItem.tsx

```diff
@@ -139,17 +139,31 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
         let header: JSX.Element = null;
```

Comment:
The `render` method is going out of control, can you refactor it into multiple smaller render methods?

```js
render() {
    if (!this.props.data.visible) {
        return null;
    }

    let iconElement: JSX.Element = renderItemIcon();

    let nestedElements: JSX.Element = null;
    let clusteredElements: JSX.Element = null;
    if (this.state.expanded && this.props.data.childItems.length > 0) {
        // Same stuff goes here...
    }

    let header: JSX.Element = renderItemHeader();
    let bodyIndentation: JSX.Element = renderBodyIndentation();
    let arrow: JSX.Element = renderItemArrow();

    return (existing stuff...);
}
```